### ISSUE TITLE
[Feature request] Add preset options on the context menu (right click)

### ISSUE THREAD

**Issue by GabeDuarteM** [n4ze3m/page-assist#163]
## Description

When downloading the extension, I was looking for something similar to Brave's Leo AI, and it does most of the things I was needing it for, but there is one which I think would be a good addition, which is adding the ability to add custom presets for when you select some text. Using Brave as an example, when you select the text and right click it, we get the following menu option:

![image](https://github.com/user-attachments/assets/439a5545-7e41-47ba-bb40-c8268a962bbb)

If I then click on one of those options, it opens the Leo AI's sidebar and automatically enters the selected text, using the selected option as part of the prompt to know what to do with that text (I assume using system prompts, but also could be done by pre-pending some preset text before the selected text), so I only need to wait for it to process and give me the answer. Ideally it would be nice to automatically send the text to the model for it to process, but it could also be nice to add an option to just "copy and paste" the text to the sidebar without sending it, so you can edit the prompt when needed

Useful default options could be predefined, like `Summarize`, `Explain`, `Rewrite`, but ideally it would be nice if we could also allow users to add custom ones as well, so we could add for example, an option to translate the selection to english, or any other user-specific need.

Can also be related to https://github.com/n4ze3m/page-assist/issues/56 to some extent, as it could also allow you to have a simpler way to just "move" the text to the page-assist's sidebar without automatically sending it, so you can add your question on the prompt after the selected text and interact with the model that way

### TLDR

1.  Similar to Brave's Leo AI, allow users to select text in a webpage and right click to get context menu options to interact with page-assist's sidebar.
2. **Options**:
   - Have some sensible predefined options: Summarize, Explain, Rewrite, etc.
   - Customizable options: Users can add their own options like 'Translate to English', etc.
3. **Behavior**: When selecting an option, it should automatically:
   - Insert the selected text into a prompt using the chosen option as context (e.g., "Summarize this: [selected text]").
   - Optionally, allow when creating custom preset options to either send the prompt for processing automatically or allow manual editing before sending.

**Comment by n4ze3m** [comment]
Hey, the latest version supports a Leo-like feature. Please note that you can edit the prompt from the 'Manage Prompt' section

![image](https://github.com/user-attachments/assets/e42456fe-8351-44da-8da3-5e433ba4ea1f)
![InnSDdu77P](https://github.com/user-attachments/assets/ea6c15cf-2383-4a10-ab45-749a25d1a12c)

**Comment by GabeDuarteM** [comment]
Awesome, exactly what I was looking for! Thank you! :rocket:

**Comment by SamuelDevdas** [comment]
> Hey, the latest version supports a Leo-like feature. Please note that you can edit the prompt from the 'Manage Prompt' section
> 
> ![image](https://private-user-images.githubusercontent.com/39720973/356213578-e42456fe-8351-44da-8da3-5e433ba4ea1f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY4Njg4NDIsIm5iZiI6MTcyNjg2ODU0MiwicGF0aCI6Ii8zOTcyMDk3My8zNTYyMTM1NzgtZTQyNDU2ZmUtODM1MS00NGRhLThkYTMtNWU0MzNiYTRlYTFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIwVDIxNDIyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2MTc2OTE3ODc2MWYyMjg4Y2FiNWE0OGJjNWQxNjM3NmI3ZWRkYzkwYThiYjhmNjUyMWQ4OWU4ZjUwNDY3MWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.34d4YuG3__V65fJJ5FXxoZXiLUlbR4H7z3PSWijrqBA) ![InnSDdu77P](https://private-user-images.githubusercontent.com/39720973/356214107-ea6c15cf-2383-4a10-ab45-749a25d1a12c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY4Njg4NDIsIm5iZiI6MTcyNjg2ODU0MiwicGF0aCI6Ii8zOTcyMDk3My8zNTYyMTQxMDctZWE2YzE1Y2YtMjM4My00YTEwLWFiNDUtNzQ5YTI1ZDFhMTJjLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIwVDIxNDIyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVkOGQzZGY5ZjUwMGI2MTQ0YWQ5NGNhYzJjOTQyMDhkNDJkOTgxMWYwODM1ZmE1NDllYmM0OTlmN2RiNTFhYzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZJRduzyOf2IM76OAwRmkixWqOqEjqmPoDUmZJuruGPw) [ ![InnSDdu77P](https://private-user-images.githubusercontent.com/39720973/356214107-ea6c15cf-2383-4a10-ab45-749a25d1a12c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY4Njg4NDIsIm5iZiI6MTcyNjg2ODU0MiwicGF0aCI6Ii8zOTcyMDk3My8zNTYyMTQxMDctZWE2YzE1Y2YtMjM4My00YTEwLWFiNDUtNzQ5YTI1ZDFhMTJjLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIwVDIxNDIyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVkOGQzZGY5ZjUwMGI2MTQ0YWQ5NGNhYzJjOTQyMDhkNDJkOTgxMWYwODM1ZmE1NDllYmM0OTlmN2RiNTFhYzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZJRduzyOf2IM76OAwRmkixWqOqEjqmPoDUmZJuruGPw) ](https://private-user-images.githubusercontent.com/39720973/356214107-ea6c15cf-2383-4a10-ab45-749a25d1a12c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY4Njg4NDIsIm5iZiI6MTcyNjg2ODU0MiwicGF0aCI6Ii8zOTcyMDk3My8zNTYyMTQxMDctZWE2YzE1Y2YtMjM4My00YTEwLWFiNDUtNzQ5YTI1ZDFhMTJjLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIwVDIxNDIyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVkOGQzZGY5ZjUwMGI2MTQ0YWQ5NGNhYzJjOTQyMDhkNDJkOTgxMWYwODM1ZmE1NDllYmM0OTlmN2RiNTFhYzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZJRduzyOf2IM76OAwRmkixWqOqEjqmPoDUmZJuruGPw) [ ](https://private-user-images.githubusercontent.com/39720973/356214107-ea6c15cf-2383-4a10-ab45-749a25d1a12c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY4Njg4NDIsIm5iZiI6MTcyNjg2ODU0MiwicGF0aCI6Ii8zOTcyMDk3My8zNTYyMTQxMDctZWE2YzE1Y2YtMjM4My00YTEwLWFiNDUtNzQ5YTI1ZDFhMTJjLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIwVDIxNDIyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVkOGQzZGY5ZjUwMGI2MTQ0YWQ5NGNhYzJjOTQyMDhkNDJkOTgxMWYwODM1ZmE1NDllYmM0OTlmN2RiNTFhYzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZJRduzyOf2IM76OAwRmkixWqOqEjqmPoDUmZJuruGPw)

Thanks for creating this amazing tool! I tried but could not find where the "custom" prompt can be changed. Would be really grateful if someone would point that out.

**Comment by n4ze3m** [comment]
Hey @SamuelDevdas  , to set a custom prompt in Page Assist, open the web UI, go to Settings, then navigate to 'Manage Prompt' and select 'Copilot Prompt.' There, you can edit the custom prompt or modify the quick prompts in the side panel.

https://github.com/user-attachments/assets/ac0031a0-a539-47c4-b4c8-984445f6ccca

**Comment by SamuelDevdas** [comment]
Thanks a lot I found it easily! I believe page-assist is really useful and has a lot of potential.

**Comment by n4ze3m** [comment]
For now, there's only one option. However, I'm checking if it's possible to add a context menu dynamically for the extension. :) Maybe it will be included in a future version
